homework/hm11_bank_card_operations.py

```diff
@@ -0,0 +1,200 @@
+"""
+HOMEWORK11. Bank card operations.
+
+Task Description:
+-----------------
+HOMEWORK11 involves the following task.
+-Write class for bank card.
+-Class should reflect card lifecycle, card operations etc.
+-Use CVV, PIN, Name, Surname , end date, card_num as initial params.
+-Class should have in addition to common logic some class attributes,
+as minimum one class method and as minimum one staticmethod,
+two or more getters/setters, __str_ magic method.
+-Do not forget about block ""if __name__ == '__main__':""
+and code there to check your class written logic.
+"""
+from datetime import datetime
+import random
+
+TERMINAL_SEPARATOR = '-' * 42
+
+
+class BankCard:
+    bank_name = "StereoBank"
+    currency = "USD"
+
+    def __init__(self, name, surname):
+        self.__balance = 0.00
+        self._is_blocked = False
+        self.name = name
+        self.surname = surname
+        self._card_number = self.generate_card_number()
+        self.__pin = self.generate_pin()
+        self.__cvv = self.generate_cvv()
+        self.__expiration_date = self.calculate_end_date()
+
+    @classmethod
+    def generate_card_number(cls):
```

Comment:
Here and below check if method has no cls inside this is static method. Classmethod we use when cls definition used by method code.